File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/tag/update/TagUpdate.java
/*
 * Copyright 2013 ETH Zuerich, CISD
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.ethz.sis.openbis.generic.asapi.v3.dto.tag.update;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.update.FieldUpdateValue;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.update.IUpdate;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.update.IdListUpdateValue;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.update.ListUpdateValue.ListUpdateAction;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.dataset.id.IDataSetId;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.experiment.id.IExperimentId;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.material.id.IMaterialId;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.sample.id.ISampleId;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.tag.id.ITagId;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author pkupczyk
 */
@JsonObject("as.dto.tag.update.TagUpdate")
public class TagUpdate implements IUpdate
{
    private static final long serialVersionUID = 1L;

    @JsonProperty
    private ITagId tagId;

    @JsonProperty
    private FieldUpdateValue<String> description = new FieldUpdateValue<String>();

    @JsonProperty
    private IdListUpdateValue<IExperimentId> experimentIds = new IdListUpdateValue<IExperimentId>();

    @JsonProperty
    private IdListUpdateValue<ISampleId> sampleIds = new IdListUpdateValue<ISampleId>();

    @JsonProperty
    private IdListUpdateValue<IDataSetId> dataSetIds = new IdListUpdateValue<IDataSetId>();

    @JsonProperty
    private IdListUpdateValue<IMaterialId> materialIds = new IdListUpdateValue<IMaterialId>();

    @JsonIgnore
    public ITagId getTagId()
    {
        return tagId;
    }

    @JsonIgnore
    public void setTagId(ITagId tagId)
    {
        this.tagId = tagId;
    }

    @JsonIgnore
    public FieldUpdateValue<String> getDescription()
    {
        return description;
    }

    @JsonIgnore
    public void setDescription(String description)
    {
        this.description.setValue(description);
    }

    @JsonIgnore
    public IdListUpdateValue<IExperimentId> getExperimentIds()
    {
        return experimentIds;
    }

    @JsonIgnore
    public void setExperimentActions(List<ListUpdateAction<IExperimentId>> actions)
    {
        experimentIds.setActions(actions);
    }

    @JsonIgnore
    public IdListUpdateValue<ISampleId> getSampleIds()
    {
        return sampleIds;
    }

    @JsonIgnore
    public void setSampleActions(List<ListUpdateAction<ISampleId>> actions)
    {
        sampleIds.setActions(actions);
    }

    @JsonIgnore
    public IdListUpdateValue<IDataSetId> getDataSetIds()
    {
        return dataSetIds;
    }

    @JsonIgnore
    public void setDataSetActions(List<ListUpdateAction<IDataSetId>> actions)
    {
        dataSetIds.setActions(actions);
    }

    @JsonIgnore
    public IdListUpdateValue<IMaterialId> getMaterialIds()
    {
        return materialIds;
    }

    @JsonIgnore
    public void setMaterialActions(List<ListUpdateAction<IMaterialId>> actions)
    {
        materialIds.setActions(actions);
    }

}

File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/tag/update/TagUpdate.js
/**
 * @author pkupczyk
 */
define([ "stjs", "as/dto/common/update/FieldUpdateValue", "as/dto/common/update/IdListUpdateValue" ], function(stjs, FieldUpdateValue, IdListUpdateValue) {
	var TagUpdate = function() {
		this.description = new FieldUpdateValue();
		this.experimentIds = new IdListUpdateValue();
		this.sampleIds = new IdListUpdateValue();
		this.dataSetIds = new IdListUpdateValue();
		this.materialIds = new IdListUpdateValue();
	};
	stjs.extend(TagUpdate, null, [], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.tag.update.TagUpdate';
		constructor.serialVersionUID = 1;
		prototype.tagId = null;
		prototype.description = null;
		prototype.experimentIds = null;
		prototype.sampleIds = null;
		prototype.dataSetIds = null;
		prototype.materialIds = null;

		prototype.getObjectId = function() {
			return this.getTagId();
		};
		prototype.getTagId = function() {
			return this.tagId;
		};
		prototype.setTagId = function(tagId) {
			this.tagId = tagId;
		};
		prototype.getDescription = function() {
			return this.description;
		};
		prototype.setDescription = function(description) {
			this.description.setValue(description);
		};
		prototype.getExperimentIds = function() {
			return this.experimentIds;
		};
		prototype.setExperimentActions = function(actions) {
			this.experimentIds.setActions(actions);
		};
		prototype.getSampleIds = function() {
			return this.sampleIds;
		};
		prototype.setSampleActions = function(actions) {
			this.sampleIds.setActions(actions);
		};
		prototype.getDataSetIds = function() {
			return this.dataSetIds;
		};
		prototype.setDataSetActions = function(actions) {
			this.dataSetIds.setActions(actions);
		};
		prototype.getMaterialIds = function() {
			return this.materialIds;
		};
		prototype.setMaterialActions = function(actions) {
			this.materialIds.setActions(actions);
		};
	}, {
		tagId : "ITagId",
		description : {
			name : "FieldUpdateValue",
			arguments : [ "String" ]
		},
		experimentIds : {
			name : "IdListUpdateValue",
			arguments : [ "IExperimentId" ]
		},
		sampleIds : {
			name : "IdListUpdateValue",
			arguments : [ "ISampleId" ]
		},
		dataSetIds : {
			name : "IdListUpdateValue",
			arguments : [ "IDataSetId" ]
		},
		materialIds : {
			name : "IdListUpdateValue",
			arguments : [ "IMaterialId" ]
		}
	});
	return TagUpdate;
})